### PR TITLE
chore: update master to angular version 8 beta

### PR DESCRIPTION
**Pull request by jelbourn**
This allows us to cherry-pick @kara's static query updates from the
`ivy-2019` branch.

This change includes:
* Adding missing dependencies for various Angular subpackages
* Add umd assets where they are now necessary
* Configure bazel to target ES5 output as a workaround for https://github.com/angular/angular/issues/29107
* Updates most npm dependencies to the @npm// workspace
* Updates general workspace setup, RBE config, ngcontainer
* Update to rxjs 6.4.0 (building from source no longer needed)
* Added rxjs_shims for ts_web_test and ts_devserver targets
* Removed index.bzl and angular_material_setup_workspace (not needed anymore since material won't be built from source)
* Use browser_repositories() from @npm_bazel_karma which have pinned browsers that work on all platforms
* Change dgeni processor functions from arrow function to `function` to
work around `@npm//di` not supporting arrow functions

cc @gregmagolan

### CODE REVIEW

File: .circleci/base-rbe-bazelrc [hunk 1]
```diff
@@ -34,7 +34,7 @@ build:remote --host_javabase=@bazel_toolchains//configs/ubuntu16_04_clang/1.1:jd
 build:remote --javabase=@bazel_toolchains//configs/ubuntu16_04_clang/1.1:jdk8
 build:remote --host_java_toolchain=@bazel_tools//tools/jdk:toolchain_hostjdk8
 build:remote --java_toolchain=@bazel_tools//tools/jdk:toolchain_hostjdk8
-build:remote --crosstool_top=@bazel_toolchains//configs/ubuntu16_04_clang/1.1/bazel_0.18.0/default:toolchain
+build:remote --crosstool_top=@bazel_toolchains//configs/ubuntu16_04_clang/1.1/bazel_0.22.0/default:toolchain
```
Comment (alexeagle) [inline .circleci/base-rbe-bazelrc]: maybe for later: I like the approach in angular/angular of vendoring one file from the bazel team, the filename is versioned with Bazel so it's easy to treat it as a black box, then you just import that file to get all these settings

File: package.json [hunk 2]
```diff
@@ -48,15 +49,17 @@
...
+    "@angular/compiler-cli": "8.0.0-beta.5",
+    "@angular/http": "8.0.0-beta.5",
+    "@angular/platform-browser-dynamic": "8.0.0-beta.5",
+    "@angular/platform-server": "8.0.0-beta.5",
+    "@angular/router": "8.0.0-beta.5",
+    "@angular/upgrade": "8.0.0-beta.5",
     "@bazel/ibazel": "^0.9.0",
-    "@bazel/karma": "0.22.1",
-    "@bazel/typescript": "0.22.1",
+    "@bazel/jasmine": "0.26.0-beta.0",
```
Comment (alexeagle) [inline package.json]: nit: shouldn't need the beta here, 0.26.0 is released

File: WORKSPACE [hunk 3]
```diff
@@ -5,49 +5,8 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
...
-# Since we still need the "npm" repository in order to use the Angular or TypeScript Bazel
-# rules, we create a local repository that is just defined in **this** workspace and is not
-# being shipped to downstream projects. This can be removed once downstream projects can
-# consume Angular Material completely from NPM.
-# TODO(devversion): remove once Angular Material can be consumed from NPM with Bazel.
-local_repository(
-  name = "npm",
-  path = "tools/npm-workspace"
+  sha256 = "86ea92217dfd4a84e1e335cc07dfd942b12899796b080492546b947f12c5ab77",
+  urls = ["https://github.com/bazelbuild/rules_nodejs/releases/download/0.26.0-beta.0/rules_nodejs-0.26.0-beta.0.tar.gz"],
```
Comment (alexeagle) [inline WORKSPACE]: nit: shouldn't need to use beta, 0.26.0 is released

File: src/lib/paginator/BUILD.bazel [hunk 4]
```diff
@@ -10,9 +10,10 @@ ng_module(
   module_name = "@angular/material/paginator",
   assets = [":paginator.css"] + glob(["**/*.html"]),
   deps = [
-    "@angular//packages/common",
-    "@angular//packages/core",
-    "@rxjs",
+    "@npm//@angular/common",
+    "@npm//@angular/core",
+    "@npm//@angular/forms", # TODO(jelbourn): transitive dep via generated code
```
Comment (devversion) [inline src/lib/paginator/BUILD.bazel]: For later reference: This is actually expected as we turned of the symbol re-exports (see https://github.com/angular/angular/commit/91b7152852e86311a9c880e558ae0caefc7e5e26).

File: tools/rxjs/rxjs_shims.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
```
Comment (devversion) [inline tools/rxjs/rxjs_shims.js]: Also just realized this. This license is not in sync with what we usually have in Material. I know that this file is from `angular/angular` and they don't use `Google LLC.`. thought just checking if we want to update it.

File: src/e2e-app/BUILD.bazel [hunk 6]
```diff
@@ -32,24 +32,42 @@ ts_devserver(
...
+      "@npm//node_modules/@angular/common:bundles/common-http.umd.js",
+      "@npm//node_modules/@angular/common:bundles/common-testing.umd.js",
+      "@npm//node_modules/@angular/common:bundles/common.umd.js",
+      "@npm//node_modules/@angular/compiler:bundles/compiler-testing.umd.js",
+      "@npm//node_modules/@angular/compiler:bundles/compiler.umd.js",
+      "@npm//node_modules/@angular/core:bundles/core-testing.umd.js",
+      "@npm//node_modules/@angular/core:bundles/core.umd.js",
+      "@npm//node_modules/@angular/forms:bundles/forms.umd.js",
+      "@npm//node_modules/@angular/router:bundles/router.umd.js",
+      "@npm//node_modules/@angular/platform-browser-dynamic:bundles/platform-browser-dynamic-testing.umd.js",
```
[the remark this answers is not in the document]
Comment (jelbourn) [reply src/e2e-app/BUILD.bazel]: I was just being lazy and copied from the testing default. I moved these out into a constant in `packages.bzl` and added a comment where it's used

File: package.json [hunk 7]
```diff
@@ -48,15 +49,17 @@
   "devDependencies": {
     "@angular-devkit/core": "7.1.2",
     "@angular-devkit/schematics": "7.1.2",
-    "@angular/bazel": "^7.2.1",
-    "@angular/compiler-cli": "^7.2.1",
-    "@angular/http": "^7.2.1",
-    "@angular/platform-browser-dynamic": "^7.2.1",
-    "@angular/platform-server": "^7.2.1",
-    "@angular/router": "^7.2.1",
+    "@angular/bazel": "angular/bazel-builds#8.0.0-beta.5+8f8f9a6",
```
[the remark this answers is not in the document]
Comment (jelbourn) [reply package.json]: Changed to beta.6 (which wasn't released when I started)

File: tools/rxjs/rxjs_shims.js (hunk 5 above)
Comment (jelbourn) [reply tools/rxjs/rxjs_shims.js]: Fixed

**Comment by josephperrott** [comment]
Correcting googlebot as all authors have CLA on file.

**Pull request merged**